pkg/appgw/internaltypes.go

```diff
@@ -230,17 +229,12 @@ func defaultBackendAddressPool(appGWIdentifier Identifier) n.ApplicationGatewayB
 func defaultFrontendListenerIdentifier() listenerIdentifier {
 	return listenerIdentifier{
 		FrontendPort: Port(80),
-		HostName:     "",
 	}
 }
 
 func (listenerID *listenerIdentifier) setHostNames(hostnames []string) {
```

Comment:
we might need some change in generateListenerID() as well in case rule.Host exisits

Reply:
That is covered here: https://github.com/Azure/application-gateway-kubernetes-ingress/blob/b03f7e953aad8fd7a2b7629cbc9762023c0a901e/pkg/appgw/configbuilder.go#L227

Reply:
> That is covered here: 
>   
>     
>       application-gateway-kubernetes-ingress/pkg/appgw/configbuilder.go
>     
>     
>          Line 227
>       in
>       b03f7e9
>     
>     
>     
>     
> 
>         
>           
>            if rule != nil && rule.Host != "" {

how about we set listenerID.HostName = "xx" here?

Reply:
is there any benefit?

Reply:
> is there any benefit?

if we don't hostname for listener anymore, I think we should be good